src/components/snapshot/SnapshotPopup.tsx

```diff
@@ -0,0 +1,247 @@
+    inline: {
+      marginRight: theme.spacing(1),
+    },
+    actions: {
+      paddingTop: theme.spacing(2),
+      paddingBottom: theme.spacing(0.5),
+      textAlign: 'end',
+    },
+    listItem: {
+      listStyleType: 'none',
+      marginBottom: theme.spacing(1),
+    },
+    bodyText: {
+      paddingBottom: theme.spacing(2),
+    },
+    light: {
+      color: 'rgba(0, 0, 0, 0.54)',
+    },
+    chip: {
+      backgroundColor: theme.palette.primary.light,
+      margin: theme.spacing(0.5),
+      marginLeft: '0px',
+    },
+    centered: {
+      textAlign: 'center',
+    },
+    exitButton: {
+      position: 'absolute',
+      top: 0,
+      right: 0,
+      padding: '1em',
+      width: '3em',
+      height: '3em',
+      cursor: 'pointer',
+      ...theme.mixins.jadeLink,
+    },
+    title: {
+      marginTop: '10px',
+      marginRight: '16px',
+      paddingBottom: '15px',
+    },
+    jadeLink: {
+      ...theme.mixins.jadeLink,
+    },
+  });
+
+interface SnapshotPopupProps {
+  classes: ClassNameMap;
+  dispatch: Dispatch<Action>;
+  filterData: any;
+  isOpen: boolean;
+  policies: Array<PolicyModel>;
+  snapshot: SnapshotModel;
+}
+
+function SnapshotPopup({
+  classes,
+  dispatch,
+  filterData,
+  isOpen,
+  policies,
+  snapshot,
+}: SnapshotPopupProps) {
+  /**
+   * When a snapshot gets created, the empty snapshot prop gets populated with a brief summary.
+   * We want to use that snapshot id to get the full object and the policies for the dialog.
+   */
+  const prevSnapshotRef = useRef<SnapshotModel>(snapshot);
+  useEffect(() => {
+    if (prevSnapshotRef.current && _.isEmpty(prevSnapshotRef.current) && !_.isEmpty(snapshot)) {
+      prevSnapshotRef.current = snapshot;
+      dispatch(
+        getSnapshotById({
+          snapshotId: snapshot.id,
+          include: [
+            SnapshotIncludeOptions.SOURCES,
+            SnapshotIncludeOptions.TABLES,
+            SnapshotIncludeOptions.RELATIONSHIPS,
+            SnapshotIncludeOptions.ACCESS_INFORMATION,
+            SnapshotIncludeOptions.PROFILE,
+            SnapshotIncludeOptions.DATA_PROJECT,
+          ],
+        }),
+      );
+      dispatch(getSnapshotPolicy(snapshot.id));
+    }
+  }, [snapshot, dispatch]);
+
+  const handleClose = () => {
+    dispatch(openSnapshotDialog(false));
+  };
+
+  const notReady = _.some([snapshot, policies, snapshot.source, snapshot.tables], _.isEmpty);
+  if (notReady) {
+    return (
+      <Dialog open={isOpen}>
+        <DialogTitle>Your data snapshot is being created</DialogTitle>
+        <DialogContent>
+          {/* TODO: Make this loading state more descriptive */}
+          <div className={clsx(classes.centered, classes.content)}>
+            <CircularProgress />
+          </div>
+        </DialogContent>
+      </Dialog>
+    );
+  }
+
+  // this number represents the total rows in the snapshot, i.e. the sum over all tables
+  const rows = snapshot.tables?.map((t) => t.rowCount).reduce((a, b) => a + b);
```

Comment:
I think this is not quite correct as `rowCount` is `number?`. Something like this should work
```suggestion
  const rows = snapshot.tables?.map((t) => t?.rowCount ?? 0).reduce((a, b) => a + b);
```

Also I think `reduce()` might return `number?` as well without an initial value, so this would let `rows` be `number`:
```
  const rows = snapshot.tables?.map((t) => t?.rowCount ?? 0).reduce((a, b) => a + b) ?? 0;
```

Reply:
ah good catch, thank you!